src/test/java/com/browserstack/ParallelTest.java

```diff
@@ -0,0 +1,58 @@
+package com.browserstack;
+
+import com.browserstack.util.Utility;
+import io.cucumber.core.cli.Main;
+import org.json.simple.JSONArray;
+import org.json.simple.JSONObject;
+import org.json.simple.parser.JSONParser;
+import org.json.simple.parser.ParseException;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.FileReader;
+import java.io.IOException;
+import java.util.Map;
+
+public class ParallelTest {
+
+    public static ThreadLocal<JSONObject> threadLocalValue = new ThreadLocal<>();
+
+    public static Logger log = LoggerFactory.getLogger(ParallelTest.class);
+
+    public static void main(String[] args) throws IOException, ParseException {
+        JSONObject testConfigs;
+        JSONObject testSelectedConfig;
+        JSONParser parser = new JSONParser();
+        if (System.getenv("caps") != null) {
+            testConfigs = (JSONObject) parser.parse(System.getenv("caps"));
+        } else {
+            testConfigs = (JSONObject) parser.parse(new FileReader("src/test/resources/config/caps.json"));
+        }
+        if (System.getProperty("caps-type") != null) {
+            testSelectedConfig = (JSONObject) ((JSONObject) testConfigs.get("tests")).get(System.getProperty("caps-type"));
+        } else {
+            testSelectedConfig = (JSONObject) ((JSONObject) testConfigs.get("tests")).get("parallel");
+        }
+        JSONArray environments = (JSONArray) testSelectedConfig.get("env_caps");
+        log.debug("Selected Test Config : " + testSelectedConfig.toJSONString());
+        for (Object obj : environments) {
```

Comment:
How can the user control the level of concurrency if the number of environments are greater than the number of parallels on their account?

Reply:
@anirudha13 - would this existing text in the README.md work or any further updates on that needed?

## Configuring the maximum parallel test threads for this repository

For all the parallel run configuration profiles, you can configure the maximum parallel test threads by changing the settings below.

- Docker

  [docker-compose.yml](docker-compose.yml)
  
  scale = 2

- BrowserStack

  Maven:

  [pom.xml](pom.xml)

  cucumber.execution.parallel.config.fixed.parallelism = 4

  Gradle:

  [build.gradle.kts](build.gradle.kts)

  cucumber.execution.parallel.config.fixed.parallelism = 4

Reply:
For parallel browsers update parallel-count configuration in pom.xml or build.gradle.kts